packages/neon/neon/lib/src/widgets/exception.dart

```diff
@@ -25,88 +25,89 @@ class NeonException extends StatelessWidget {
+    final color = this.color ?? Theme.of(context).colorScheme.error;
 
-              return Column(
-                mainAxisAlignment: MainAxisAlignment.center,
-                children: [
-                  Row(
-                    mainAxisAlignment: MainAxisAlignment.center,
-                    children: [
-                      errorIcon,
-                      const SizedBox(
-                        width: 10,
-                      ),
-                      Flexible(
-                        child: Text(
-                          details.text,
-                          style: TextStyle(
-                            color: color ?? Colors.red,
-                          ),
-                        ),
-                      ),
-                    ],
+    final errorIcon = Icon(
+      Icons.error_outline,
+      size: iconSize ?? 30,
+      color: color,
+    );
+
+    final message =
+        details.isUnauthorized ? AppLocalizations.of(context).loginAgain : AppLocalizations.of(context).actionRetry;
+
+    if (onlyIcon) {
+      return Semantics(
+        tooltip: details.text,
+        child: IconButton(
+          icon: errorIcon,
+          padding: EdgeInsets.zero,
+          visualDensity: const VisualDensity(
+            horizontal: VisualDensity.minimumDensity,
+            vertical: VisualDensity.minimumDensity,
+          ),
+          tooltip: message,
+          onPressed: () async {
+            if (details.isUnauthorized) {
+              _openLoginPage(context);
+            } else {
+              onRetry();
+            }
+          },
+        ),
+      );
+    }
+
+    return Padding(
+      padding: const EdgeInsets.all(5),
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: [
+          Row(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: [
+              errorIcon,
+              const SizedBox(
+                width: 10,
+              ),
+              Flexible(
+                child: Text(
+                  details.text,
+                  style: TextStyle(
+                    color: color,
                   ),
-                  if (details.isUnauthorized) ...[
-                    ElevatedButton(
-                      onPressed: () async => _openLoginPage(context),
-                      child: Text(AppLocalizations.of(context).loginAgain),
-                    ),
-                  ] else ...[
-                    ElevatedButton(
-                      onPressed: onRetry,
-                      child: Text(AppLocalizations.of(context).actionRetry),
-                    ),
-                  ],
-                ],
-              );
+                ),
+              ),
+            ],
+          ),
+          ElevatedButton(
+            onPressed: () async {
+              if (details.isUnauthorized) {
+                _openLoginPage(context);
+              } else {
+                // TODO: why no retry?
```

Comment:
The old implementation did not call onRetry in this case but the iconButton does.
Is this intended?

Reply:
I'm probably blind as the old implementation clearly calls onRetry :woman_facepalming: